model.py

```diff
@@ -1364,6 +1364,17 @@ class ExternalIntegration(Base):
     INTERNAL_LOGGING = u'Internal logging'
     LOGGLY = u'Loggly'
 
+    # If there is a special URL to use for access to this API,
+    # put it here.
+    URL = u"url"
+
+    # If access requires authentication, these settings represent the
+    # username/password or key/secret combination necessary to
+    # authenticate. If there's a secret but no key, it's stored in
+    # 'password'.
+    USERNAME = u"username"
+    PASSWORD = u"password"
+
```

Comment:
All this code is taken from core. The tests didn't catch it because setting `ExternalIntegration.url` works fine in Python without this code, and the test passes -- the data just isn't stored in the database.